models/exposure.py

```diff
@@ -352,28 +439,19 @@ def update_instrument(self, session=None):
         with SmartSession(session) as session:
             self.instrument_object.fetch_sections(session=session, dateobs=self.mjd)
 
-    def calculate_coordinates(self):
-        if self.ra is None or self.dec is None:
-            raise ValueError("Exposure must have RA and Dec set before calculating coordinates! ")
-
-        coords = SkyCoord(self.ra, self.dec, unit="deg", frame="icrs")
-        self.gallat = coords.galactic.b.deg
-        self.gallon = coords.galactic.l.deg
-        self.ecllat = coords.barycentrictrueecliptic.lat.deg
-        self.ecllon = coords.barycentrictrueecliptic.lon.deg
+    @staticmethod
+    def _do_not_require_file_to_exist():
+        """
+        By default, new Exposure objects are generated
+        with nofile=False, which means the file must exist
+        at the time the Exposure object is created.
+        This is the opposite default from the base class
+        FileOnDiskMixin behavior.
+        """
+        return False
 
 
 if __name__ == '__main__':
-
-    from models.base import Session
-    import models.instrument
-
-    import numpy as np
-    rnd_str = lambda n: ''.join(np.random.choice(list('abcdefghijklmnopqrstuvwxyz'), n))
-
-    e = Exposure(f"Demo_test_{rnd_str(5)}.fits", exp_time=30, mjd=58392.0, filter="F160W", ra=123, dec=-23, project='foo', target='bar')
-
-    session = Session()
-
-    session.add(e)
-    session.commit()
+    filename = '/home/guyn/Dropbox/python/SeeChange/data/DECam_examples/c4d_221104_074232_ori.fits.fz'
```

Comment:
I realize this is something we're never going to run (we'll import exposure, not python it), but still it's a little ugly to have a hardcoded path on your machine here....

I have, of course, done exactly this sort of thing.  But, when I'm behaving better, what I try to do is create a subdirectory called "muckabout" or something like that where I can, well, muck about.  There I have the little hardcoded test thingies that I run on my own machine to make sure to be able to look into what's going on.

Reply:
I think I left that in by accident... I'll fix it right up